src/MergePlugin.php

```diff
@@ -135,6 +144,21 @@ public static function getSubscribedEvents()
     }
 
     /**
+     * Handle an event callback for initialization.
+     *
+     * @param \Composer\EventDispatcher\Event $event
+     */
+    public function onInit(\Composer\EventDispatcher\Event $event)
+    {
+        $this->state->loadSettings();
+        // @todo devMode cannot be properly detected at this stage, but
+        //       merging it should not hurt.
+        $this->state->setDevMode(true);
```

Comment:
I'm not sure that this assumption is really safe. Unconditionally enabling dev mode merging will make it impossible for anyone using Composer >=v1.1.0 to omit merged dev mode packages using the `--no-dev` CLI argument. This feels like a "but why would anyone do ..." rationalization that is easy to make during initial coding and difficult to fix later when someone presents a real world use case. The problem with not setting it to true is that currently `mergeFile` only merges each file one time to avoid certain side effects of double loading. This introduces the inverse problem of never being able to manage dev mode requirements if the flag is not set initially.

A possible fix would be to split merging in some way so that differentiates between processing in the two different dev mode states. Done properly this could allow the INIT event to process everything with `setDevMode(false)` (which is always safe and wanted) and then depending on the command invoked possibly trigger another pass with `setDevMode(true)` based on the known state when an install, update, or dump event is processed.